### ISSUE TITLE
pcap2pandas(): KeyError for 'datetime'

### ISSUE THREAD

**Issue by Rameen-Mahmood** [noise-lab/netml#31]
Encountering an issue with the `pcap2pandas()` method in the `netml.pparser.parser` module. When a pcap file other than the provided demo.pcap is used, a `KeyError: 'datetime'` is raised. 

**Code:**
```
from netml.pparser.parser import PCAP

pcap = PCAP('path/to/your.pcap')
pcap.pcap2pandas()

pdf = pcap.df
```
**Error:**
```
Exception has occurred: KeyError
'datetime'
KeyError: 'datetime'

The above exception was the direct cause of the following exception:

File "path/to/your_script.py", line X, in <module>
    pcap.pcap2pandas()
KeyError: 'datetime'
```

**Environment:**

- netml v0.6.0
- python 3.11.9

**Comment by Rameen-Mahmood** [comment]
https://drive.google.com/file/d/116a6glE2LR6HmCMfuNPhaZvX5RTkCDSP/view?usp=sharing

**Comment by jesteria** [comment]
OK. None of the packets in that PCAP appear to contain any ethernet device information; but, this method was written to only consider packets which do contain ethernet device information.

Certainly, the method can be adjusted to handle this edge case, at least to provide a more useful error message.

Beyond that, do you expect this method to generate a `DataFrame` for this stream? I don't personally see why not; it would only be missing MAC address source and destination information. Nonetheless, I'd be curious to hear the thoughts of the author, @feamster, (judging by [the record](https://github.com/noise-lab/netml/commit/3e564164#diff-d450004fa4d6abbc7dd88fb4e04ea12ca824374fd1cb4ff704fe4d59bb1c6c24R974-R975)).

**Comment by crazyideas21** [comment]
Turns out we don't have the MAC layer because we captured traffic using a WireGuard VPN (that's our method of getting mobile traffic). 

@Rameen-Mahmood will try to disable MAC layer check and add dummy fields into the MAC layer. In our case, the MAC layer is boring because it only shows either the gateway or the phone.

Thank you @jesteria for the pointer to the code!

**Comment by jesteria** [comment]
If it helps, you might try the fix proposed by #32 – that is, in branch `jsl/fix-dataframe-no-mac`.

**Comment by jesteria** [comment]
FYI: The fix has been merged and the above-mentioned branch has been deleted.
